src/test/java/integration/BaseTest.java

```diff
@@ -15,35 +15,81 @@
+        username = System.getenv("APISERVER_USERNAME");
+        password = System.getenv("APISERVER_PASSWORD");
+        baseUrl = System.getenv("APISERVER_REPOSITORY_API_BASE_URL");
+        authorizationType = System.getenv("AUTHORIZATION_TYPE");
         String testHeaderValue = System.getenv("TEST_HEADER");
-        if (spKey == null && accessKeyBase64 == null && repoId == null && testHeaderValue == null) {
+        if (nullOrEmpty(authorizationType)) {
             // Load environment variables
             Dotenv dotenv = Dotenv
                     .configure()
                     .filename(".env")
                     .load();
-            accessKeyBase64 = dotenv.get("ACCESS_KEY");
-            spKey = dotenv.get("SERVICE_PRINCIPAL_KEY");
-            repoId = dotenv.get("REPOSITORY_ID");
+            authorizationType = dotenv.get("AUTHORIZATION_TYPE");
             testHeaderValue = dotenv.get("TEST_HEADER");
+            repoId = dotenv.get("REPOSITORY_ID");
+            if (nullOrEmpty(repoId)) {
+                throw new IllegalStateException("Environment variable REPOSITORY_ID does not exist.");
+            }
+            if (authorizationType.equalsIgnoreCase(AuthorizationType.CLOUD_ACCESS_KEY.name())) {
+                if (nullOrEmpty(spKey) && nullOrEmpty(accessKeyBase64)) {
+                    accessKeyBase64 = dotenv.get("ACCESS_KEY");
+                    spKey = dotenv.get("SERVICE_PRINCIPAL_KEY");
+                    accessKey = AccessKey.createFromBase64EncodedAccessKey(accessKeyBase64);
+                }
+            } else if (authorizationType.equalsIgnoreCase(AuthorizationType.API_SERVER_USERNAME_PASSWORD.name())) {
+                if (nullOrEmpty(username) && nullOrEmpty(password) && nullOrEmpty(baseUrl)) {
+                    username = dotenv.get("APISERVER_USERNAME");
+                    password = dotenv.get("APISERVER_PASSWORD");
+                    baseUrl = dotenv.get("APISERVER_REPOSITORY_API_BASE_URL");
+                }
+            } else {
+                throw new IllegalStateException("Invalid Authorization Type Value");
+            }
         }
-        accessKey = AccessKey.createFromBase64EncodedAccessKey(accessKeyBase64);
         testHeaders = new HashMap<>();
         testHeaders.put(testHeaderValue, "true");
-        repositoryApiClient = RepositoryApiClientImpl.CreateFromAccessKey(spKey, accessKey);
-        repositoryApiClient.setDefaultRequestHeaders(testHeaders);
+        repositoryApiClient = createClient();
+    }
+
+    public static RepositoryApiClient createClient() {
+        if (repositoryApiClient == null) {
+            if (authorizationType.equalsIgnoreCase("Cloud_Access_Key")) {
+                if (nullOrEmpty(spKey) || accessKey == null)
+                    return null;
+                repositoryApiClient = RepositoryApiClientImpl.createFromAccessKey(spKey, accessKey);
+            } else if (authorizationType.equalsIgnoreCase("API_Server_Username_Password")) {
```

Comment:
should use the enum instead of hardcode string `"API_Server_Username_Password"`. Same for above `"Cloud_Access_Key"` on line 80

Reply:
ok done